tt-metal build should work with shallow clone

### Describe the bug

I'm running `tt-metal` build as part of CI. And I clone it with regular `actions/checkout@v4`, which does shallow clone (which is completely reasonable). It worked fine since recently (about a week ago it worked fine).

But now I get the error:
```
CMake Error at /usr/local/share/cmake-3.28/Modules/WriteBasicConfigVersionFile.cmake:43 (message):
  No VERSION specified for WRITE_BASIC_CONFIG_VERSION_FILE()
Call Stack (most recent call first):
  /usr/local/share/cmake-3.28/Modules/CMakePackageConfigHelpers.cmake:239 (write_basic_config_version_file)
  cmake/packaging.cmake:45 (write_basic_package_version_file)
  CMakeLists.txt:315 (include)
```

`tt-metal` tries to determine its version, but as it's shallow clone, it fails.

I would expect that `tt-metal` would default to some pre-defined value, instead of failing. I obviously don't care about the version in this environment and I don't actually run packaging.

### Steps to reproduce the issue

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error


### Expected behavior

_No response_

### Please complete the following environment information

_No response_

--- a/cmake/version.cmake
+++ b/cmake/version.cmake
@@ -30,9 +30,15 @@ function(ParseGitDescribe)
             ERROR_QUIET
         )
     endif()
+    if(NOT VERSION_HASH)
+        set(VERSION_HASH ${fallbackHash})
+    endif()
     if(NOT version)
         set(version ${fallbackVersion})
-        set(VERSION_HASH ${fallbackHash})
+        # A shallow Git clone will fail a git describe, but also will not have substitued the fallbackVersion
+        if(version MATCHES "Format")
+            set(version "0.0-alpha0-1-g${VERSION_HASH}-dirty")
+        endif()
     endif()
 
     # Local modifications (dirty), or not

